modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java

```diff
@@ -446,17 +441,13 @@ null, _getLogoBytes(account, null, false), account.getTaxId(),
 			_getType(account), _getStatus(account),
 			_createServiceContext(account));
 
-		if (_isValidId(account.getDefaultBillingAddressId())) {
-			_accountEntryLocalService.updateDefaultBillingAddressId(
-				accountEntry.getAccountEntryId(),
-				account.getDefaultBillingAddressId());
-		}
+		_accountEntryLocalService.updateDefaultBillingAddressId(
+			accountEntry.getAccountEntryId(),
+			_getDefaultBillingAddress(account));
```

Comment:
I think we should because there is no further null check or anything in updateDefaultBillingAddressId() that would prevent updating it if _getDefaultBillingAddress provides 0 or null and it will because in case of post, there is no default value in case of addresses.